config/routes.rb

```diff
@@ -1,4 +1,10 @@
 Rails.application.routes.draw do
   get '/', to: 'application#welcome'
   # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
+
+  #merchant invoices ---------------------------------
+  namespace :merchants do
+    get '/:merchant_id/invoices', to: 'invoices#index'
+    get '/:merchant_id/invoices/:invoice_id', to: 'invoices#show'
```

Comment:
Your route paths look good!  You might consider checking out nested resources.

Check out the section on nested resource:  https://guides.rubyonrails.org/v5.2/routing.html#nested-resources

For example, you could do:
```
resources :merchant do
  resources :invoices, only: [:index, :show]
end
```

And this will generate routes with prefixes you can use for path helpers.

Reply:
very cool! I'll look into this when Caroline and I merge our routes together :)